rmw_opensplice_cpp/src/rmw_service_names_and_types.cpp

```diff
@@ -119,7 +120,17 @@ rmw_get_service_names_and_types(
       // Duplicate and store each type for the service
       size_t type_index = 0;
       for (const auto & type : service_n_types.second) {
-        char * type_name = rcutils_strdup(type.c_str(), *allocator);
+        size_t n = type.find(SAMPLE_PREFIX);
+        if (std::string::npos == n) {
+          RMW_SET_ERROR_MSG_ALLOC(
+            "failed to convert DDS type name to ROS service type name: '"
+            SAMPLE_PREFIX "' not found",
```

Comment:
Sorry, another late note: maybe printing the actual `type` here would make the error message more useful.